### ISSUE TITLE
Update Visual Studio 2022 to 17.8

### ISSUE THREAD

**Issue by Sergio0694** [actions/runner-images#8797]
### Tool name

Visual Studio 2022

### Tool license

https://visualstudio.microsoft.com/license-terms/vs2022-ga-proenterprise/

### Add or update?

- [ ] Add
- [X] Update

### Desired version

17.8 (https://visualstudio.microsoft.com/en/downloads/)

### Brief description of tool

This is the latest stable release of VS2022, which includes the .NET 8 SDK.

It's also explicitly needed to build Desktop-MSBuild-dependent project with .NET 8.

### Platforms where you need the tool

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [X] Windows Server 2022

**Comment by Sergio0694** [comment]
Hey @shamil-mubarakshin, would you be able to help with this, and/or provide an ETA for when this will be available? Because GitHub Actions don't have preview versions of VS, updating to 17.8 is needed to unblock building projects that depend on Desktop MSBuild and that also need the .NET 8 SDK (right now, they'd only find the .NET 7 SDK that way, and fail).

Thank you! 🙂

**Comment by Sergio0694** [comment]
Yup that's completely fair! That's also why I only opened this now that VS 17.8 just went live 😄

Thank you for your help, looking forward to the next image updates!

**Comment by stevenvolckaert** [comment]
We're using Azue Pipelines hosted agent `windows-2022`.

Is this the correct issue for updating the included Visual Studio version to version 17.8? Or should I open a new issue for that?

**Comment by Sergio0694** [comment]
@Alexey-Ayupov @mikhailkoliada just trying to better plan the rest of my work that depends on this, do you know when is the deployment of the new Windows2022 image with VS 17.8 planned for? I see #8826 is already up as well with the changelog 👀

Thank you!

**Comment by Marv51** [comment]
@Sergio0694 I'm just lurking here, cause I am also waiting for 17.8. 

On the main page it says under "Rollout Progress of Latest Image Release", for windows-2022 and 20231115.2.1
13.65%.

 [![statuswin22](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&badge=1)](https://gh-runnerimagesdeploymentstatus.azurewebsites.net/api/status?imageName=windows-2022&redirect=1)

So I would interpret that to mean that it is already rolling out?

**Comment by Sergio0694** [comment]
Ooh interesting, I had missed that! I had just tried running my pipeline again, and it still failed due to picking up the image with VS 17.7 still, so I thought it was just not available at all yet. Is there a way to opt-in into using the latest image that's rolling out, or is that just randomly assigned when a new job goes to pick up an image to run on? Ie. do I just re-trigger it until it works? 😄

**Comment by JanRajnoha** [comment]
What about availability in Azure DevOps? Will it be available at same time as in GitHub Actions?

**Comment by xmarshal** [comment]
We also waiting Azure vmimage with .net 8.0 support

**Comment by steve-torchia** [comment]
Yesterday, mid-day, our image seems to have been changed to include .net 8.0 support but it BROKE our build which was depending on the previous image.   Is there any way for us to get back to that previous image?   This was a pretty big braking change on us

**Comment by asd-oak** [comment]
All of our Azure DevOps pipelines, which ask for `windows-latest`, picked up the revised `windows-2022` image at about this time yesterday. We had .net8 branches triggering (failing) builds until this showed up; after it did, all* subsequent builds got the new image. I don't know if we got in early because we had failing builds or were just lucky, but I'll take it.

\* One pipeline's run one time got the old image. Retry of the job got the new image.

**Comment by steve-torchia** [comment]
Our problem is that the revised windows-2022 image started failing our builds because the newer version of MSBuild was raising Nuget Warnings as errors.    In our case we need to be able to use that older image so we dont have those failures.   Basically our builds started breaking across our entire company due to the underlying image changing - not due to something we did on our own

**Comment by gtbuchanan** [comment]
I can confirm most Azure Pipelines `windows-2022` agents we've seen in the wild today have been updated to VS 17.8. Unfortunately, this came with a regression of the Java SDK being downgraded and breaking Xamarin Android builds (see #8837 and https://github.com/microsoft/azure-pipelines-agent/issues/4531).

**Comment by Sergio0694** [comment]
Still can't pick up VS 17.8 in any of my pipeline runs. And the rollout percentage seems to be... Going back down? 🥲

**Comment by JanRajnoha** [comment]
I tried pipelines and some works for 1st attempt (so i thought image is deployed), but one has to be tried for nine times to pick version with .NET 8 :/

**Comment by autometrist** [comment]
I don't know if I've been very lucky to get the updated image right after adding this to my workflow, but it's worked since (and rather fast too, for a VS Update?)

```
      - run: Invoke-WebRequest https://aka.ms/vs/17/release/vs_enterprise.exe -OutFile C:/Temp/vs_enterprise.exe
      - run: C:/Temp/vs_enterprise.exe --update --quit --wait
      - run: C:/Temp/vs_enterprise.exe update --wait --passive --norestart --installPath "C:\Program Files (x86)\Microsoft Visual Studio\2022\Enterprise"

```

**Comment by taspeotis** [comment]
Just add this?

```yaml
- name: Install WinGet
  uses: Cyberboss/install-winget@v1

- run: winget install -e --silent --accept-source-agreements --accept-package-agreements --id Microsoft.VisualStudio.2022.Enterprise
  continue-on-error: true
```

50 minutes later:

![image](https://github.com/actions/runner-images/assets/1189399/ccb6e477-9448-43a1-8562-9e738dcdaba0)

**Comment by Sergio0694** [comment]
That doesn't really seem viable though if you have a larger pipeline, if it takes about 50 minutes to install VS2022 in each job you have. Eg. my pipeline has 12 of them 🥲

**Comment by Sergio0694** [comment]
Looks like rollout is nearing 90%, and I can see all of my pipelines consistently picking up the new image now 🎉

Thanks everyone working on the runner images for the quick turnaround!

**Comment by mikhailkoliada** [comment]
Deployed to all standard runners

**Comment by nevercast** [comment]
Just as a heads up, this seems to have broken my Azure Pipeline builds, as MSBuild 17.8.3 is unable to restore packages from private repos, whereas MSBuild 17.7.3 had no such issue. I suspect I won't be the only one and further issues will be raised.
